control-plane/connect-inject/common/common.go

```diff
@@ -116,3 +119,40 @@ func ShouldIgnore(namespace string, denySet, allowSet mapset.Set) bool {
 func ConsulNodeNameFromK8sNode(nodeName string) string {
 	return fmt.Sprintf("%s-virtual", nodeName)
 }
+
+// ********************
+// V2 Exclusive Common Code
+// ********************
+
+// ToProtoAny is a convenience function for converting proto.Message values to anypb.Any without error handling.
+// This should _only_ be used in cases where a nil or valid proto.Message value is _guaranteed_, else it will panic.
+func ToProtoAny(m proto.Message) *anypb.Any {
+	if m == nil {
+		return nil
+	}
+	a, err := anypb.New(m)
+	if err != nil {
+		panic(fmt.Errorf("unexpected error: failed to convert proto message to anypb.Any: %w", err))
+	}
+	return a
+}
+
+// GetPortProtocol matches the Kubernetes EndpointPort.AppProtocol or ServicePort.AppProtocol (*string) to a supported
+// Consul catalog port protocol. If nil or unrecognized, the default of `PROTOCOL_UNSPECIFIED` is returned.
+func GetPortProtocol(appProtocol *string) pbcatalog.Protocol {
+	if appProtocol == nil {
+		return pbcatalog.Protocol_PROTOCOL_UNSPECIFIED
+	}
+	switch *appProtocol {
+	case "tcp":
+		return pbcatalog.Protocol_PROTOCOL_TCP
+	case "http":
+		return pbcatalog.Protocol_PROTOCOL_HTTP
+	case "http2":
+		return pbcatalog.Protocol_PROTOCOL_HTTP2
+	case "grpc":
+		return pbcatalog.Protocol_PROTOCOL_GRPC
```

Comment:
I wonder if we technically need to include prefixes with some of these protocols. I was reading the API docs for [appProtocol](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#serviceport-v1-core) and it mentions only[ IANA names](https://www.iana.org/assignments/service-names) can be unprefixed. Only my first time seeing this so I could be mistaken.

I also think an annotation on the service or pod might be easier for customers to support. e.g. the [Bitnami Postgres helmchart](https://github.com/bitnami/charts/tree/main/bitnami/postgresql) exposes annotations for services and certain pods but not a change of appProtocol.

Reply:
Added this as a bullet in follow-up ticket so we don't lose track but probably don't need to figure out right this second.

Reply:
Circling back to this one to tie up the loose end: @ishustava, do you happen to have perspective on either of Dan's questions above?

My gut is to leave the unprefixed protocols in place for now since there seems to be precedent in other meshes, and wait to see if we need annotations in addition to just `appProtocol`. If that's not ideal, we can fix now or plan to follow up in 1.18.